src/game/level/level_editor/point_layer.c

```diff
@@ -488,28 +500,22 @@ int point_layer_edit_id_event(PointLayer *point_layer,
     case SDL_KEYDOWN: {
         switch(event->key.keysym.sym) {
         case SDLK_RETURN: {
-            char *ids = dynarray_data(point_layer->ids);
-            const char *text = edit_field_as_text(point_layer->edit_field);
+            RenameContext rename_context = create_rename_context(
+                point_layer,
+                (size_t) point_layer->selected);
 
-            Action action = {
-                .revert = point_layer_revert_rename,
-                .layer = point_layer
-            };
-
-            ASSERT_CONTEXT_SIZE(RenameContext);
-            RenameContext *rename_context = (RenameContext *)action.context.data;
-
-            memcpy(
-                rename_context->id,
-                ids + point_layer->selected * ID_MAX_SIZE,
-                ID_MAX_SIZE);
-            rename_context->index = (size_t) point_layer->selected;
-
-            undo_history_push(undo_history, action);
+            undo_history_push(
+                undo_history,
+                create_action(
+                    point_layer,
+                    point_layer_revert_rename,
+                    &rename_context,
+                    sizeof(rename_context)));
 
-            size_t n = max_size_t(strlen(text), ID_MAX_SIZE - 1);
-            memcpy(ids + point_layer->selected * ID_MAX_SIZE, text, n);
-            *(ids + point_layer->selected * ID_MAX_SIZE + n) = '\0';
+            char *id = dynarray_pointer_at(point_layer->ids, (size_t) point_layer->selected);
+            const char *text = edit_field_as_text(point_layer->edit_field);
+            memset(id, 0, ID_MAX_SIZE);
```

Comment:
```suggestion
            int n = min_size_t(strlen(text), ID_MAX_SIZE - 1);
            memset(id, text, n);
            memset(id + n, 0, ID_MAX_SIZE - n);
```

This is a bit more optimised.

Reply:
memset will set memory content of id to the pointer value of text instead of copying the content pointed to by text.

Reply:
```suggestion
            strncpy(id, text, ID_MAX_SIZE - 1);
            id[ID_MAX_SIZE - 1] = '\0';
```